src/actions/MultiCall.sol

```diff
@@ -0,0 +1,58 @@
+/**
+ * Created by Arcadia Finance
+ * https://www.arcadia.finance
+ *
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+pragma solidity >0.8.13;
+
+import "../actions/utils/ActionData.sol";
+import "../interfaces/IERC20.sol";
+import "../interfaces/IERC1155.sol";
+
+/**
+ * @title Generic multicall action
+ * @author Arcadia Finance
+ * @notice Call any external contract with arbitrary data. Return the balances of assets that need to be deposited within a vault.
+ * @dev Only calls are used, no delegatecalls
+ * This address will approve random addresses. Do not store any funds on this address!
+ */
+
+contract ActionMultiCall {
+    /**
+     * @notice Calls a series of addresses with arbitrrary calldata
+     * @dev input address is not used in this generic action.
+     * @param actionData a bytes object containing two actionAssetData structs, an address array and a bytes array
+     * @return incoming a actionAssetData struct with the balances of this ActionMultiCall address.
+     */
+    function executeAction(address, bytes calldata actionData) external returns (actionAssetsData memory) {
```

Comment:
Will we ever need the first input variable address as input?

Reply:
hm maybe we might need it on a action handler for a specific usecase (like LP'ing or airdrop claims)?

Reply:
But it has to be triggered in the Vault anyway?
Because first you need to withdraw?